tools/dune_rule_gen/util.ml

```diff
@@ -0,0 +1,42 @@
+(* LICENSE NOTE: This file is dually MIT/LGPL 2.1+ licensed. MIT license:
```

Comment:
Why?

Reply:
As explained in the main file header, this implementation is taken from Dune's own coq_rules, in fact they are very close, so once Dune has a plugin API we could unify them. This was the plan we discussed a while back.

Reply:
Oh, for that particular file, the story is a bit more special, we used to have stuff from OCaml's stdlib, now indeed we still have one function from 4.10 Stdlib, so here we have to use this license.

Reply:
What I wrote before applies to the other files (args, coq_module, coq_rules, etc...)

Reply:
Dually licensed means that users can take the file and decide what license to apply. If there is code from OCaml inside then the file needs to be licensed under LGPL-2.1-only. It *cannot* be dual licensed with the MIT license and it *cannot* be LGPL-2.1+. Furthermore, the copyright header must be preserved from the OCaml source file. Given that the file seems to only contain one function, just keep the header from the OCaml distribution and do not add anything else.

Reply:
or rewrite that one function, that is clearly the simplest option

Reply:
Indeed the license is wrong as this is taken from https://github.com/ocaml/ocaml/blob/trunk/stdlib/list.ml , will amend the license which seems even simpler to me.

Reply:
The mess of licenses here is due to the old tormented history of this coq_dune development.